### PR TITLE
App Publisher UI : Basic Layout.

### PR DESCRIPTION
**Pull request by menakaj**

### CODE REVIEW

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/pom.xml [hunk 1]
```diff
@@ -0,0 +1,169 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Megala21) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/pom.xml:1]: Please add the license header

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/pom.xml [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+                                    <propertyDef>org.wso2.carbon.p2.category.type:server
+                                    </propertyDef>
+                                    <propertyDef>org.eclipse.equinox.p2.type.group:false
+                                    </propertyDef>
+                                </properties>
+                            </adviceFile>
+                            <bundles>
+
+                            </bundles>
+                            <importFeatures>
```
Comment (Megala21) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/pom.xml]: Why do we have empty blocks? Please remove them

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/pom.xml [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+        </profile>
+    </profiles>
+    <properties>
+        <maven.test.skip>false</maven.test.skip>
+        <npm.executable>npm</npm.executable>
+        <npm.build.command>build_prod</npm.build.command>
+        <npm.working.dir>./src/main/</npm.working.dir>
+    </properties>
+    
+</project>
```
Comment (Megala21) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/pom.xml]: Add an extra line at the end of the file.

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/pom.xml [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+                    <!--<phase>test</phase>-->
+                    <!--<configuration>-->
+                    <!--<workingDirectory>${basedir}/src/main/resources/publisher</workingDirectory>-->
+                    <!--<skip>${maven.test.skip}</skip>-->
+                    <!--<executable>${npm.executable}</executable>-->
+                    <!--<arguments>-->
+                    <!--<argument>test</argument>-->
+                    <!--</arguments>-->
+                    <!--</configuration>-->
+                    <!--</execution>-->
```
Comment (Megala21) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/pom.xml]: Why do we have commented code ? Please clear them

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/public/index.html [hunk 5]
```diff
@@ -0,0 +1,40 @@
+<!doctype html>
```
Comment (Megala21) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/public/index.html:1]: Don't we need the license?

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/App.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+import './App.css'
+import React, {Component} from 'react';
+import createHistory from 'history/createHashHistory';
+import {HashRouter as Router, Redirect, Route, Switch} from 'react-router-dom'
+import {BaseLayout, Create, Login, NotFound, PublisherOverview} from './components'
+
+const history = createHistory({basename: '/publisher'});
+
+class Base extends Component {
```
Comment (Megala21) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/App.js]: Can we add comments to indicate what is the purpose of each component, if React supports that ?

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/actions/ApplicationActions.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
+/*
```
Comment (Megala21) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/actions/ApplicationActions.js:1]: What is the purpose of having files only with license header? Are they needed ?

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/Application/Forms/Step1.js [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+
+    handlePrev() {
+        this.props.handlePrev();
+    }
+
+    onChangePlatform = (event, index, value) => {
+        this.setState({platform: value});
+    };
+
+    onChangeStore = (event, index, value) => {
```
Comment (Megala21) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/Application/Forms/Step1.js]: I think it is better to add come comments for user-defined functions.

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/Application/Forms/Step3.js [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+                            onClick={this.handleFinish.bind(this)}
+                        />
+                    </div>
+                </div>
+            </div>
+        );
+    }
+}
+
+export default Step3;
```
Comment (Megala21) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/Application/Forms/Step3.js]: Please add an extra line at the end of the file. Please change it in other places as well.

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/webpack.config.js [hunk 10]
```diff
@@ -0,0 +1,51 @@
+var path = require('path');
```
Comment (Megala21) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/webpack.config.js:18]: Missing License header

File: components/application-mgt/org.wso2.carbon.device.application.mgt.store.ui/pom.xml [hunk 11]
```diff
@@ -0,0 +1,168 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Megala21) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.store.ui/pom.xml:1]: License header is missing here too.

File: components/application-mgt/org.wso2.carbon.device.application.mgt.store.ui/pom.xml [hunk 12]
```diff
@@ -0,0 +1,168 @@
...
+        <artifactId>application-mgt</artifactId>
+        <version>3.0.46-SNAPSHOT</version>
+    </parent>
+    <artifactId>org.wso2.carbon.device.application.mgt.store.ui</artifactId>
+    <version>3.0.46-SNAPSHOT</version>
+
+    <dependencies>
+        <!--<dependency>-->
+        <!--<groupId>org.wso2.carbon.apimgt</groupId>-->
+        <!--<artifactId>org.wso2.carbon.apimgt.publisher.feature</artifactId>-->
```
Comment (Megala21) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.store.ui/pom.xml]: Please remove the commented code

File: components/application-mgt/org.wso2.carbon.device.application.mgt.store.ui/pom.xml [hunk 13]
```diff
@@ -0,0 +1,168 @@
...
+                        <phase>initialize</phase>
+                        <configuration>
+                            <workingDirectory>${basedir}/src/main/resources/store</workingDirectory>
+                            <executable>${npm.executable}</executable>
+                            <arguments>
+                                <argument>install</argument>
+                            </arguments>
+                        </configuration>
+                    </execution>
+                    <!-- 'npm run clean' which is run in clean lifecycle needs 'rimraf' node module.
```
Comment (Megala21) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.store.ui/pom.xml]: Why do we have commented code?

File: features/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui.feature/pom.xml [hunk 14]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Megala21) [inline features/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui.feature/pom.xml:1]: License header is missing

**Review by Megala21 — CHANGES_REQUESTED** [review]

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/pom.xml (hunk 2 above)
Comment (menakaj) [reply components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/pom.xml]: Removed the empty blocks.

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/public/index.html (hunk 5 above)
Comment (menakaj) [reply components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/public/index.html:1]: License added.

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/actions/ApplicationActions.js (hunk 7 above)
Comment (menakaj) [reply components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/actions/ApplicationActions.js:1]: Added the base content for the JS files.

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/pom.xml [hunk 15]
```diff
@@ -0,0 +1,131 @@
...
+
+                            </bundles>
+                            <importFeatures>
+
+                            </importFeatures>
+                            <importBundles>
+
+                            </importBundles>
+                        </configuration>
+                    </execution>
```
Comment (Megala21) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/pom.xml]: bundles, importBundles are still empty

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/public/index.html [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+<!--
+~ Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (Megala21) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/public/index.html]: License year is wrong

**Pull request merged**